app/renderer/components/Inspector/GestureEditor.js

```diff
@@ -0,0 +1,565 @@
+      const x = parseFloat(updateX, 10);
+      const y = parseFloat(updateY, 10);
+      if (coordType === COORD_TYPE.PERCENTAGES) {
+        currentTick.x = this.pixelsToPercentage(x, true);
+        currentTick.y = this.pixelsToPercentage(y, false);
+      } else {
+        currentTick.x = x;
+        currentTick.y = y;
+      }
+      this.setState({pointers});
+    }
+  }
+
+  // Actions for adding/deleting pointers
+  pointerAction (targetKey, action) {
+    const {unselectTick} = this.props;
+    const {pointers} = this.state;
+
+    if (action === ACTION_TYPES.ADD) {
+      const key = this.state.pointers.length + 1;
+      const newKey = String(key);
+      pointers.push({
+        name: `pointer${key}`,
+        ticks: [{id: `${key}.1`}],
+        id: newKey,
+        color: COLORS[key - 1]
+      });
+      this.setState({activeKey: newKey, pointers});
+    } else {
+      // newKey variable keeps track of the previous pointer before deleting the current one
+      let newKey = '1';
+      const filteredPointers = pointers.filter((pointer) => pointer.id !== targetKey);
+      const newPointers = filteredPointers.map((pointer, index) => {
+        const id = String(index + 1);
+        if (id !== pointer.id) {
+          pointer.id = id;
+          pointer.color = COLORS[index];
+          pointer.ticks = (pointer.ticks).map((tick) => {
+            tick.id = `${id}.${tick.id[2]}`;
+            return tick;
+          });
+        } else {
+          newKey = pointer.id;
+        }
+        return pointer;
+      });
+      unselectTick();
+      this.setState({activeKey: newKey, pointers: newPointers});
+    }
+  }
+
+  // Actions for adding/deleting ticks
+  tickAction (pointerKey, tickKey, action) {
+    const {unselectTick} = this.props;
+    const {pointers} = this.state;
+    const currentPointer = pointers.find((pointer) => pointer.id === pointerKey);
+
+    if (action === ACTION_TYPES.ADD) {
+      const id = `${pointerKey}.${(currentPointer.ticks).length + 1}`;
+      currentPointer.ticks.push({id});
+    } else {
+      const filteredTicks = currentPointer.ticks.filter((tick) => {
+        if (tick.id === tickKey) {
+          unselectTick();
+        } else {
+          return tick;
+        }
+      });
+      const newTicks = filteredTicks.map((tick, index) => {
+        const id = String(index + 1);
+        if (tick.id !== id) {
+          tick.id = `${tick.id[0]}.${id}`;
+        }
+        return tick;
+      });
+      currentPointer.ticks = newTicks;
+    }
+    this.setState({pointers});
+  }
+
+  // Updates the current tick within local state
+  updateTick (tick, msg, value) {
+    const {POINTER_TYPE} = TICK_PROPS;
+    const {selectTick} = this.props;
+    const {pointers} = this.state;
+    const currentPointer = pointers.find((pointer) => pointer.id === tick.id[0]);
+    currentPointer.ticks = (currentPointer.ticks).map((prevTick) => {
```

Comment:
The reason why we can't mutate the object as it is because React doesnt allow you to mutate state objects. So if were mutating a complex object like this gesture within the local state, then we have to send a whole new copy of the previous object + any changes to the local set state so it can be saved within the state. Its really weird! However, I can apply your change above but it leads to a whole bunch of `if` statements, I thought this way would be a lot more cleaner and it doesnt require us to to type check on a bunch of values, and also allows us to restructure each tick if we are changing a pointer type. Else we would have to use a delete operator and write a lot of unnecessary code, for some reason some of the lodash functions dont operate properly when were set state and etc.

Reply:
ok, I get it about using setState. just mutating the object doesn't trigger a state update.

i'm not getting your point about how it would look more complicated using my suggestion, however. it would actually involve fewer if statements than you currently have, the way i see it? (it would add one if statement that you don't currently have, in the case where we don't find a tick that matches the requested tick; but this is a case which isn't  handled by your current code and might be a potential bug)